core/src/test/java/apoc/periodic/PeriodicTest.java

```diff
@@ -78,6 +78,28 @@ public void testSubmitStatement() throws Exception {
         testCall(db, callList, (r) -> assertEquals(true, r.get("done")));
     }
 
+    @Test
+    public void testSubmitStatementWithParams() throws Exception {
+        String callList = "CALL apoc.periodic.list()";
+        // force pre-caching the queryplan
+        assertFalse(db.executeTransactionally(callList, Collections.emptyMap(), Result::hasNext));
+
+        testCall(db, "CALL apoc.periodic.submit('foo','create (:Foo { id: $id })', { id: '(╯°□°)╯︵ ┻━┻' })",
```

Comment:
The params passed to this query are `{ id: '(╯°□°)╯︵ ┻━┻' }`

And the test verifies that a node with that ID exists, meaning that the only way this could pass is if the params are taken & respected, as the ID doesn't appear in the query (only by param mention)

Reply:
WDYT of aligning this parameter configuration with https://neo4j.com/labs/apoc/4.1/overview/apoc.periodic/apoc.periodic.iterate/? In that case, the parameters are passed under the key `params`. For instance (thanks @JMHReif!):

```cypher
CALL apoc.periodic.iterate(‘MATCH (m:Movie) WHERE m.title IN $movies RETURN m’,
‘WITH m CALL apoc.load.json($prefix+m.movieId+$suffix) YIELD value
UNWIND value.cast AS cast
MERGE (c:Cast {id: cast.id})
  ON CREATE SET c.name = cast.name
MERGE (ch:Character {name: cast.character})
MERGE (ch)-[r:APPEARS_IN]->(m)
MERGE (c)-[r1:PLAYED]->(ch)’,
{batchSize: 1, iterateList:false, params:{movies:movies, prefix:prefix, suffix:suffix}})
```

Reply:
@fbiville at present there are no other "options" to pass to this call.  The style of using params as a subkey makes sense when params are one of many options that can be passed (batchSize, iterateList, etc).

If there is only one thing you'd ever pass (params) then nesting "params" as an object key under another object doesn't make sense in my opinion.

So this really boils down to (for me) a design question of whether you think you want to introduce other controlling options to this call in the future.

FWIW -- if you wish (for consistency reasons) to create a certain kind of "configuration object" which acts as a common last parameter to APOC calls, then this might should be described in the documentation.  You could separate from any one APOC call document what a "config" object is, and then in the docs for any proc, simply indicate which keys are permitted.   This could be valuable for overall consistency in APOC.   But honestly I'm not sure what other kinds of config we'd want on this call

More broadly, this is a point change to an area that feels in need of re-factoring.  For example, the reason that I wanted apoc.periodic.submit in the background is specifically because it's a one-off background job.  The proc isn't very well named for that purpose right now (imo).

Reply:
@moxious, I think this makes sense to us. Seems we misread some code. Everything is good as-is.

We're happy to rename the proc! If you have ideas for a better name (a developer's worst enemy - naming things), feel free to toss an issue our way. :)

Reply:
@JMHReif this is only a semi-serious suggestion; personally I find the apoc.periodic stuff a bit misnamed, or differently named than what I'd expect as a user.  You could collapse apoc.periodic.submit and apoc.periodic.iterate into one call if you wanted with an easier name and different options.  This is only a semi-serious suggestion though because users are so used to it as it is:


```
CALL apoc.job.submit({
   background:  (true|false),
   iterateWith:  "statement_per_item",
   query: STRING
   parallel: (true|false)
   batchSize: INTEGER,
   params: MAP,
   concurrency: INTEGER,
   retries: INTEGER
})
```

So:

```apoc.period.iterate('MATCH (n) RETURN n', "DETACH DELETE n', { parallel: false, batchSize: 1000 })```

Would become:

```
apoc.job.submit({
   query: "MATCH (n) RETURN n",
   iterateWith: "DETACH DELETE n",
   batchSize: 1000,
   parallel: false,
   background: false
})
```

And 

```apoc.periodic.submit("MATCH (u:User { id: $id }) SET lastUpdated=datetime()", { id: 1 })```

Would become

```
apoc.job.submit({
   query: "MATCH (u:User { id: $id }) SET lastUpdated=datetime()",
   params: { id: 1 },
   background: true
})
```